scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,525 @@
+        gpiiKeyRecords, options.snapsetPrefsSafes
+    );
+    fluid.log("\tGPII Keys associated with snapset Prefs Safes marked for deletion.");
+    return options.gpiiKeys;
+};
+
+/**
+ * Given all the GPII Keys records in the database, find the ones that reference
+ * a snapset PrefsSafe.  As each GPII Key is found it is marked for
+ * deletion.
+ * @param {Array} gpiiKeyRecords - Array of GPII Key records from the database.
+ * @param {Array} snapSets - Array of snapset Prefs Safes whose id references
+ *                           its associated GPII Key record.
+ * @return {Array} - the values from the gpiiKeyRecords that are snapset GPII Keys.
+ */
+gpii.dataLoader.markPrefsSafesGpiiKeysForDeletion = function (gpiiKeyRecords, snapSets) {
+    var gpiiKeysToDelete = [];
+    fluid.each(gpiiKeyRecords.rows, function (gpiiKeyRecord) {
+        var gpiiKey = fluid.find(snapSets, function (aSnapSet) {
+            if (gpiiKeyRecord.value.prefsSafeId === aSnapSet._id) {
+                return gpiiKeyRecord.value;
+            }
+        }, null);
+        if (gpiiKey !== null) {
+            gpiiKey._deleted = true;
+            gpiiKeysToDelete.push(gpiiKey);
+        }
+    });
+    return gpiiKeysToDelete;
+};
+
+/**
+ * Utility to wrap all the pieces to make a bulk documents deletion request
+ * for the snapset Prefs Safes and their associated GPII keys.  Its intended use
+ * is the parameter of the appropriate promise.then() call.
+ * @param {Object} batchDeleteResponse - the reponse handler configured for the
+ *                                       batch delete request.
+ * @param {Object} options - Object that contains the records to be deleted.
+ * @return {http.ClientRequest} - The http request object.
+ */
+gpii.dataLoader.configureBatchDelete = function (batchDeleteResponse, options) {
+    var docsToRemove = options.snapsetPrefsSafes.concat(options.gpiiKeys);
+    return gpii.dataLoader.createPostRequest(
+        docsToRemove, batchDeleteResponse, options
+    );
+};
+
+/**
+ * Create a function that makes a bulk docs POST request using the given data.
+ * @param {Object} dataToPost - JSON data to POST and process in bulk.
+ * @param {Object} responseHandler - http response handler for the request.
+ * @param {Object} options - Data loader options, specifically the POST options.
+ * @return {http.ClientRequest} - An http request object.
+ */
+gpii.dataLoader.createPostRequest = function (dataToPost, responseHandler, options) {
+    var batchPostData = JSON.stringify({"docs": dataToPost});
+    options.postOptions.headers["Content-Length"] = Buffer.byteLength(batchPostData);
+    var batchDocsRequest = http.request(options.postOptions, responseHandler);
+    batchDocsRequest.write(batchPostData);
+    return batchDocsRequest;
+};
+
+/**
+ * Generate a response handler, setting up the given promise to resolve/reject
+ * at the correct time.
+ * @param {Function} handleEnd - Function to call that deals with the response
```

Comment:
Document signature of this function, possibly using an @callback JSDocs directive

Reply:
See below.